src/util/issue.ts

```diff
@@ -31,11 +31,16 @@ export async function adaptLocations(flow: Flow) {
 
 export function createDiagnosticFromIssue(issue: protocol.Issue) {
   const { startLine, startLineOffset, endLine, endLineOffset } = issue.textRange;
+  let startPosition = new vscode.Position(0, 0);
+  let endPosition  = new vscode.Position(0, 0);
+  let range = new vscode.Range(startPosition, endPosition);
   // vscode line positions are 0-based
-  const startPosition = new vscode.Position(startLine - 1, startLineOffset);
-  const endPosition = new vscode.Position(endLine - 1, endLineOffset);
-  const range = new vscode.Range(startPosition, endPosition);
-
+  if (startLine !== 0 && endLine !== 0) {
+    // this is a file-level issue
```

Comment:
If I understand correctly, this is the opposite: the issue should be considered at the file level if either `startLine` or `endLine` is missing.

Maybe this could be extracted to a function, e.g `getSafeTextRange`.

Also, it could be unit-tested 😇